catalog/spatial/csw/spatial-csw-endpoint/src/main/java/org/codice/ddf/spatial/ogc/csw/catalog/endpoint/CswQueryFactory.java

```diff
@@ -180,21 +174,11 @@ private CswRecordMapperFilterVisitor buildFilter(QueryConstraintType constraint,
                     CswConstants.NO_APPLICABLE_CODE,
                     null);
         }
-        visitor.setVisitedFilter(filter);
-        boolean isTypeNameCsw = typeNames.contains(new QName(CswConstants.CSW_OUTPUT_SCHEMA,
-                CswConstants.CSW_RECORD_LOCAL_NAME,
-                CswConstants.CSW_NAMESPACE_PREFIX));
-        boolean isTypeNameGmd = typeNames.contains(new QName(GmdMetacardType.GMD_NAMESPACE,
-                GmdMetacardType.GMD_LOCAL_NAME));
-
-        if (isTypeNameCsw || isTypeNameGmd) {
-            try {
-                visitor.setVisitedFilter((Filter) filter.accept(visitor, null));
-            } catch (UnsupportedOperationException ose) {
-                throw new CswException(ose.getMessage(),
-                        CswConstants.INVALID_PARAMETER_VALUE,
-                        null);
-            }
+
+        try {
+            visitor.setVisitedFilter((Filter) filter.accept(visitor, null));
```

Comment:
In order to maintain compliance with the CITE we must validate the typeNames parameter.
https://github.com/opengeospatial/ets-csw202/blob/master/src/main/scripts/ctl/GetRecords/CSW-GetRecords-POST.xml#L1871